src/components/input/input.jsx

```diff
@@ -1,37 +1,65 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
 import styles from './input.css';
-import labelPropType from './label-prop-type';
 
-/**
- * Base input component. Accepts any prop that an input can take.
- * @see [RFC](https://github.com/mavenlink/rfc/pull/75)
- * @see [Design spec](https://zpl.io/295Yp7p)
- */
-export default function Input({ 'aria-labelledby': ariaLabelledBy, className, id, ...rest }) {
+function getLabelClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-label'] : styles.label;
+}
+
+function getClassName(className, invalid) {
+  if (className) return className;
+  return invalid ? styles['invalid-input'] : styles.input;
+}
+
+export default function Input(props) {
   return (
-    <input
-      {...rest}
-      aria-labelledby={ariaLabelledBy}
-      className={className}
-      id={id}
-    />
+    <React.Fragment>
+      <label htmlFor={props.id} className={getLabelClassName(props.cssLabel, props.invalid)}>
+        {props.label}
+      </label>
+      <div className={styles.container}>
+        <input
+          className={getClassName(props.className, props.invalid)}
+          disabled={props.disabled}
+          id={props.id}
+          name={props.name}
+          onChange={props.onChange}
+          placeholder={props.placeholder}
+          required={props.required}
+          type="text"
+          value={props.value}
+        />
+        {props.invalid && <Icon className={styles['invalid-icon']} currentColor="caution" name={cautionSvg.id} />}
```

Comment:
Do we want to standardize on this specific ordering of invalid messages? Or, allow the consumer of MDS to handle positioning such things themselves? The later seems consistent with our approach of almost always supporting `{ children }` where it makes sense.

Reply:
UPDATE: I don't think this is yet fully finalized, but again, after speaking with Nathan at standup today, I think he's fine with having some _opinion_ and forcing matters when we want to. So, I think we still have some questions to answer here, but nothing blocking per se.

Reply:
Yup, we're going to stick with the caution icon being on the right side.